### PR TITLE
Pre-fuel type re-factor

### PR DESCRIPTION
**Pull request by Sybrand**
I started implementing setting fuel type, but ran into some very confusing code. This re-factor addresses some of the issues (code all over the place where it doesn't belong)

I don't know how much sense it makes, but I've created a number of things under *app.fire_behaviour" that makes some sense to me.

# Test Links:
[Percentile Calculator](https://wps-pr-1913.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1913.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1913.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1913.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1913.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1913.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1913.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1913.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: api/app/db/crud/hfi_calc.py [hunk 1]
```diff
@@ -25,8 +25,20 @@ def get_all_stations(session: Session) -> CursorResult:
     return session.query(PlanningWeatherStation.station_code).all()
 
 
+def get_fire_centre_station_codes() -> List[int]:
```
Comment (Sybrand) [inline api/app/db/crud/hfi_calc.py:28]: This used to be in utils - but I think it belongs right here in the crud layer. Asking for a list of station codes from the database.

File: api/app/hfi/hfi_calc.py [hunk 2]
```diff
@@ -17,18 +18,102 @@
...
                                   get_fire_weather_stations, get_fire_centre_fire_start_ranges,
                                   get_fire_start_lookup)
 
 logger = logging.getLogger(__name__)
 
 
+def generate_station_daily(raw_daily,  # pylint: disable=too-many-locals
+                           station: WFWXWeatherStation,
+                           fuel_type: FuelType) -> StationDaily:
+    """ Transform from the raw daily json object returned by wf1, to our daily object.
```
Comment (Sybrand) [inline api/app/hfi/hfi_calc.py:37]: This used to be in wfwx, but has all kinds of things that go beyond WFWXWeatherStation slapped on, specific to the hfi response, so I've thrown it in here.

File: api/app/tests/hfi/test_hfi_endpoint_request.py [hunk 3]
```diff
@@ -90,9 +90,3 @@ def given_hfi_calc_url(monkeypatch: pytest.MonkeyPatch, url: str, verb: str):
 @then(parsers.parse("request == saved = {request_saved}"), converters={'request_saved': strtobool})
 def then_request_saved(spy_store_hfi_request: MagicMock, request_saved: bool):
     assert spy_store_hfi_request.called == request_saved
-
-
-@then("the response isn't cached")
```
Comment (Sybrand) [inline api/app/tests/hfi/test_hfi_endpoint_request.py:95]: This was duplicated elsewhere - removed it.

**Review by conbrad — APPROVED** [review]

File: api/app/db/crud/hfi_calc.py [hunk 4]
```diff
@@ -78,3 +90,8 @@ def get_fire_centre_fire_start_ranges(session: Session, fire_centre_id: id) -> C
 def get_fire_start_lookup(session: Session) -> CursorResult:
     """ Get the fire start lookup table """
     return session.query(FireStartLookup)
+
+
+def get_fuel_type(session: Session, fuel_type_id: int) -> FuelType:
```
Comment (andrea-williams) [inline api/app/db/crud/hfi_calc.py:95]: could we slightly rename this function to `get_fuel_type_by_id`?

Comment (andrea-williams) [reply api/app/db/crud/hfi_calc.py:95]: I don't know if it's considered a best practice or not, but I just like it when the function name makes it clear what attribute the lookup is based on.

File: api/app/fire_behaviour/advisory.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        self.flame_length = flame_length
+        self.sixty_minute_fire_size = sixty_minute_fire_size
+        self.sixty_minute_fire_size_t = None
+        self.thirty_minute_fire_size = thirty_minute_fire_size
+        self.critical_hours_hfi_4000 = critical_hours_hfi_4000
+        self.critical_hours_hfi_10000 = critical_hours_hfi_10000
+        self.hfi_t = hfi_t
+        self.ros_t = ros_t
+        self.cfb_t = cfb_t
+        self.sixty_minute_fire_size_t = sixty_minute_fire_size_t
```
Comment (andrea-williams) [inline api/app/fire_behaviour/advisory.py:86]: could we please document somewhere what all the `_t`s mean? Off the top of my head I don't know what they're for, or how `hfi_t` differs from `hfi`

File: api/app/fire_behaviour/advisory.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+                 ros_t: Optional[float],
+                 cfb_t: Optional[float],
+                 sixty_minute_fire_size_t: Optional[float]):
+        self.hfi = hfi
+        self.hfi_t = None
+        self.ros = ros
+        self.ros_t = None
+        self.fire_type = fire_type
+        self.cfb = cfb
+        self.cfb_t = None
```
Comment (andrea-williams) [inline api/app/fire_behaviour/advisory.py:76]: Why are the `_t` values being assigned None here, when a few lines below they're being assigned the value passed to init?

File: api/app/fire_behaviour/prediction.py [hunk 7]
```diff
@@ -166,93 +89,6 @@ def calculate_cfb(fuel_type: FuelTypeEnum, fmc: float, sfc: float, ros: float, c
...
-    return FireBehaviourAdvisory(
-        hfi=hfi, ros=ros, fire_type=fire_type, cfb=cfb, flame_length=flame_length,
-        sixty_minute_fire_size=sixty_minute_fire_size,
-        thirty_minute_fire_size=thirty_minute_fire_size,
-        critical_hours_hfi_4000=critical_hours_4000,
-        critical_hours_hfi_10000=critical_hours_10000,
-        hfi_t=hfi_t, ros_t=ros_t, cfb_t=cfb_t, sixty_minute_fire_size_t=sixty_minute_fire_size_t)
-
-
 def get_fire_size(fuel_type: FuelTypeEnum, ros: float, bros: float, ellapsed_minutes: int, cfb: float,
```
Comment (andrea-williams) [inline api/app/fire_behaviour/prediction.py:92]:
```suggestion
def get_fire_size(fuel_type: FuelTypeEnum, ros: float, bros: float, elapsed_minutes: int, cfb: float,
```

Comment (andrea-williams) [reply api/app/fire_behaviour/prediction.py:92]: as long as we're refactoring... elapsed is spelled wrong ;)

**Review by andrea-williams — APPROVED** [review]
I have no strong feelings either way regarding this refactor. I don't think it's perfect and I'd like to see us do more refactoring at some point, but I think this is an improvement over our existing codebase, and I'm happy with taking the baby-steps approach to refactoring.

**Comment by andrea-williams** [comment]
Some of the Python unit tests have failed, so pretend I didn't just approve this PR ;)

**Review by andrea-williams — CHANGES_REQUESTED** [review]
Unit tests need fixin'.

File: api/app/fire_behaviour/advisory.py (hunk 5 above)
Comment (Sybrand) [reply api/app/fire_behaviour/advisory.py:83]: Sure - I'll do that in my next PR if that's ok? (I'm honestly not EXACTLY sure what it means)

Comment (andrea-williams) [reply api/app/fire_behaviour/advisory.py:83]: @Sybrand  ?

**Review by andrea-williams — APPROVED** [review]

**Pull request merged**